app/utility/RepDataMap.js

```diff
@@ -63,25 +63,35 @@ export const isValidData = (repData) => {
         // v3 bulk data flag check
```

Comment:
- [ ] isValidData needs unit tests, especially for inf / negativfe as there were too many changes in logic.